src/olympia/abuse/cinder.py

```diff
@@ -107,12 +113,21 @@ def id(self):
         return str(self.user.id)
 
     def get_attributes(self):
-        return {
+        data = {
             'id': self.id,
             'name': self.user.display_name,
             'email': self.user.email,
             'fxa_id': self.user.fxa_id,
         }
+        if self.user.picture_type and backup_storage_enabled():
+            filename = copy_file_to_backup_storage(
```

Comment:
is there a check we can make before this, to ensure we're not uploading multiple times?  (check a database field, etc).  Just seems unexpected that repeated calling `get_attributes` - that builds an attribute dict - would also hash and upload a file.

Reply:
`copy_file_to_backup_storage()` will avoid the upload if the target filename already exists. We're still hashing and making a request every time though yeah... Not sure how to avoid that.

The problem is, if we instead generate a backup copy every time the user changes their photo, and store the backup name to then send it when it gets reported, we have to potentially keep the backup copy forever instead of limiting its life to ~ 6 months.

Maybe I could, as a follow-up, store the hash on change instead, so that it doesn't have to be regenerated everytime ? That would still leave the request to potentially store the backup copy. I'm not sure it's worth the effort.

Reply:
storing the hash (or an update datetime?) would mean we wouldn't be attempted a remote call at least (because I guess the check to see if the target filename exists involves at least one call to google storage)

Reply:
Just storing the hash doesn't avoid the remote calls - we still need to make the remote call to ensure the file exists or upload it if it doesn't, and another to create the signed URL